sdk/core/Azure.Core/src/Shared/OperationInternal.cs

```diff
@@ -231,15 +180,15 @@ private set
         /// <param name="cancellationToken">A <see cref="CancellationToken"/> controlling the request lifetime.</param>
         /// <returns>The last HTTP response received from the server, including the final result of the long-running operation.</returns>
         /// <exception cref="RequestFailedException">Thrown if there's been any issues during the connection, or if the operation has completed with failures.</exception>
-        public async ValueTask<Response<T>> WaitForCompletionAsync(TimeSpan pollingInterval, CancellationToken cancellationToken)
+        public async ValueTask<Response> WaitForCompletionAsync(TimeSpan pollingInterval, CancellationToken cancellationToken)
```

Comment:
But in the case that WaitForCompletion would call this internal method instead of the one that returns OperationState<T> and not set `Value`?

https://github.com/Azure/azure-sdk-for-net/blob/c5f840df810a9017cf597f6b21bfd84251e875c5/sdk/core/Azure.Core/src/Shared/OperationInternal.cs#L233

Reply:
Can we make it virtual an override in `OperationInternal<T>` to do the right thing?